### PR TITLE
BIP Claims Evidence Final Integration

### PR DESCRIPTION
**Pull request by au2**
## What was the problem?

- Claims Evidence API was not fully integrated.

Associated tickets or Slack threads:

- [MCP-2009](https://amida.atlassian.net/browse/MCP-2009)

## How does this fix it?[^1]

This PR finalizes the Claims Evidence API integration

- Payload and Header fields in call the BIP CE finalized. We are using FILENUMBER for the E-Folder folder. Payload fIeld values discussed in slack discussions updated.
- Services for BIP API and BIP Evidence API are separated. This will enable to select in service mocks for one without effecting other.
- No temporary file is save anymore. Byte[] is directly send.
- Also moved the primary bean RestTemplate. RestTemplate for CE API was causing problems

## How to test this PR
There is not an easy way to test to locally since the Mock Claims Evidence API is not fully integrated to docker compose yet and otherwise BIP API's are not locally availabkle. But the following what I will do. If it is too much just look at the code.
- Make sure application builds and tests out of the box in local.
- Set ENV=dev. Three other environment variables should be set. BIP_PASSWORD to keystore_pw, BIP_KEYSTORE to value in https://github.com/department-of-veterans-affairs/abd-vro/blob/develop/mock-bip-ce-api/src/test/resources/client-keystore.yml, BIP_TRUSTSTORE to value in https://github.com/department-of-veterans-affairs/abd-vro/blob/develop/mock-bip-ce-api/src/test/resources/truststore.yml.
- Remove Service and conditional from MockBIPCEService. Remove conditional from BIPCEService.
- Build the application,
- Run the mock service in a new terminal: ./gradlew :mock-bip-ce-api:bootRun
- Run the positive MAS end2end test or exercise the MAS end point with the collection Id there.
- Look at the terminal. It should have recieved the payload and file. The file will be save to a temp file. Take a look at the file to see if it is the expected PDF.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: service/provider/src/main/java/gov/va/vro/service/provider/BipApiProps.java [hunk 1]
```diff
@@ -37,4 +45,27 @@ public class BipApiProps {
   private String applicationId;
 
   private String applicationName;
+
+  /**
+   * Creates common Jwt claims.
+   *
+   * @return Claims
+   */
+  public Claims toCommonJwtClaims() {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/BipApiProps.java:54]: another type of claim!

**Review by dimitri-amida — APPROVED** [review]
lgtm

**Review by wlin2019 — APPROVED** [review]
I left some comments. I haven't tested, especially for providerData passing to BIP CE. Reading from the code, I think it is okay.

File: service/provider/src/main/java/gov/va/vro/service/provider/BipApiProps.java [hunk 2]
```diff
@@ -37,4 +45,27 @@ public class BipApiProps {
...
+    Claims claims = Jwts.claims();
+    claims.put("jti", jti);
+    claims.put("applicationID", applicationId);
+    claims.put("stationID", stationId);
+    claims.put("userID", claimClientId);
+    Date now = cal.getTime();
+    claims.put("iat", now.getTime());
+    claims.put("expires", expired.getTime());
+    Map<String, Object> headerType = new HashMap<>();
+    headerType.put("typ", Header.JWT_TYPE);
```
[the remark this answers is not in the document]
Comment (au2) [reply service/provider/src/main/java/gov/va/vro/service/provider/BipApiProps.java:67]: Not sure. Just copied what you had.

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java [hunk 3]
```diff
@@ -182,51 +179,29 @@ public FetchPdfResponse uploadPdf(MasAutomatedClaimPayload payload, FetchPdfResp
...
-    }
+
+    byte[] decoder = Base64.getDecoder().decode(pdfResponse.getPdfData());
+    BipFileProviderData providerData =
+        BipFileProviderData.builder()
+            .contentSource("VRO")
+            .claimantFirstName(payload.getFirstName())
+            .claimantLastName(payload.getLastName())
+            .claimantSsn(payload.getVeteranIdentifiers().getSsn())
+            .documentTypeId(1489)
```
[the remark this answers is not in the document]
Comment (au2) [reply service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipClaimService.java:190]: OK. Will do that later.

**Pull request merged**